File: examples/java/kora-java-resilient/src/main/java/ru/tinkoff/kora/example/resilient/RetryService.java
package ru.tinkoff.kora.example.resilient;

import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.resilient.retry.annotation.Retry;

@Component
public class RetryService {

    @Retry("custom1")
    public String getValue(String arg) {
        return arg;
    }
}

File: examples/kotlin/kora-kotlin-resilient/src/main/kotlin/ru/tinkoff/kora/kotlin/example/resilient/RetryService.kt
package ru.tinkoff.kora.kotlin.example.resilient

import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.common.annotation.Root
import ru.tinkoff.kora.resilient.circuitbreaker.annotation.CircuitBreaker
import ru.tinkoff.kora.resilient.fallback.annotation.Fallback
import ru.tinkoff.kora.resilient.retry.annotation.Retry
import ru.tinkoff.kora.resilient.timeout.annotation.Timeout
import java.util.concurrent.ThreadLocalRandom

@Component
open class RetryService {
    @Retry("custom1")
    open fun getValue(arg: String): String = arg
}

